File: prince-of-versions/src/main/java/co/infinum/princeofversions/ApplicationConfiguration.java
package co.infinum.princeofversions;

/**
 * Class provides application's parameters:
 * <ul>
 * <li>version name</li>
 * <li>device SDK version code</li>
 * </ul>
 */
public interface ApplicationConfiguration {

    /**
     * Method provides application's version code
     *
     * @return application version code
     */
    int version();

    /**
     * Method provides device's SDK version code
     *
     * @return SDK version code
     */
    int sdkVersionCode();

}

File: prince-of-versions/src/main/java/co/infinum/princeofversions/ApplicationConfiguration.kt
package co.infinum.princeofversions

/**
 * Provides application parameters such as version and SDK level.
 */
interface ApplicationConfiguration {
    /**
     * The application's version code.
     */
    val version: Int

    /**
     * The device's SDK version code.
     */
    val sdkVersionCode: Int
}
